conans/client/rest/rest_client_v1.py

```diff
@@ -311,7 +311,7 @@ def remove_packages(self, ref, package_ids=None):
         payload = {"package_ids": package_ids}
         url = self.router.remove_packages(ref)
         ret = self._post_json(url, payload)
-        if package_ids is None and ret.status_code == 404:
+        if package_ids is not None and ret.status_code == 404:
```

Comment:
I don't understand this change. The error in the server happens when we try to remove all the packages, and we don't specify ``package_ids``. If we explictly specify ``-p=<package_id>``, then, the 404 error is expected and desired.